### DISCUSSION TITLE
Reverse Proxy without SWAG?

### DISCUSSION THREAD

**Discussion by PanavisionT16** [mealie-recipes/mealie#3982] [Q&A]
Is it possible to do a reverse proxy without using SWAG? All my other reverse proxies are managed through my kemp load balancer (hardware) and then routed through my domain DNS at cloudflare. How would I need to configure my yaml file to send everything over 443 to avoid the Invalid HTTP request? Should I use a tunnel with cloudflare? Any help would be appreaciated!

**Comment by boc-the-git** [comment]
You need _something_ to handle redirecting traffic via HTTPS.. the common way this is done in the self hosting community is via a reverse proxy. Any of them should work.

SWAG is not a requirement for Mealie..

**Reply by PanavisionT16** [reply]
I tried using a cloudflare tunnel but I keep getting a bad gateway error. I'm mapping to http://localhost:9925. Am I missing something?

**Reply by PanavisionT16** [reply]
ok, I figured it out. Instead of mapping to localhost, I mapped it to the IP address of the host:
![image](https://github.com/user-attachments/assets/451193c5-1eb3-4f34-999e-a7406ba21c23)
